src/js/abstract/base-component.js

```diff
@@ -176,11 +184,29 @@ export default class BaseComponent extends HTMLElement {
       lifecycleLogger(this.logLifecycle)(`+++ attributeChangedCallback -> ${this.nodeName}#${this._id} | ${name} from ${oldValue} to ${newValue}\n`);
     }
 
+    // only update the value if it has actually changed
+    // and only re-render if it has changed
+    if (!this.shouldComponentUpdate(newValue, oldValue)) {
+      return;
+    }
+
     const key = camelize(name);
 
     this[key] = toProp(newValue);
   }
 
+  /**
```

Comment:
@LucaMele 
This suuuper important! It makes sure that the component only re-renders if any attribute has actually changed:)
And it fixes the nested `<input>` issue afore mentioned 🎉